### ISSUE TITLE
Lack of disabled attribute on select element causing color contrast issue

### ISSUE THREAD

**Issue by mellis481** [JedWatson/react-select#5550]
When `isDisabled` is set to `true`, the `disabled` attribute is correctly added to the `<input>` element that makes up react select.

https://codesandbox.io/s/codesandboxer-example-forked-dsc99j

![image](https://user-images.githubusercontent.com/39313898/215495474-53ba956b-a124-40fd-97b4-1ae993f42edc.png)

Unfortunately, the actual select element (or rather the `div` that represents the select) is not properly identified as disabled and, as a result, tools like Accessibility Insights for Web complain that color contrast requirements are not met. Color contrast requirements do not need to be met for disabled controls, but tools are not seeing the select as disabled because it is not marked up properly.

![image](https://user-images.githubusercontent.com/39313898/215495296-45bc6b1d-ff34-4ad6-85a1-7299d84e8e92.png)

This can be solved by added `aria-disabled="true"` to the `control--is--disabled` node. 

![image](https://user-images.githubusercontent.com/39313898/215496164-dd5d47b7-98b9-4d9a-960b-e4a8b34c312a.png)
